src/User/Controller/SettingsController.php

```diff
@@ -347,9 +354,43 @@ public function actionNetworks()
         );
     }
 
+    public function actionConsent()
+    {
+        $user = Yii::$app->user->identity;
+
+        if (!!$user->gdpr_consent)
+            $this->redirect(['profile']);
+
+        $model = new DynamicModel(['gdpr_consent']);
+        $model->addRule('gdpr_consent', 'boolean');
+        $model->addRule('gdpr_consent', 'default', ['value' => 0, 'skipOnEmpty' => false]);
+        $model->addRule('gdpr_consent', 'compare', [
+            'compareValue' => true,
+            'message' => Yii::t('usuario', 'Your consent is required to work with this site'),
```

Comment:
this message should probably be customizable too

Reply:
but it's probably easily achievable by [overriding the translation message](https://lorenzo.mile.si/yii2-quick-tip-overriding-core-translations/648/), just needs to be documented